src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/DeferredResolver.scala

```diff
@@ -17,10 +17,15 @@ class DeferredResolver extends sangria.execution.deferred.DeferredResolver[Consi
         consignmentId.map(
           id => context.consignmentService.getConsignmentFileProgress(id)
         ).getOrElse(Future.successful(0))
+      case DeferParentFolder(consignmentId) =>
+        consignmentId.map(
+          id => context.consignmentService.getConsignmentParentFolder(id)
+        ).getOrElse(Future.successful(0))
       case other => throw UnsupportedDeferError(other)
     }
   }
 }
 
 case class DeferTotalFiles(consignmentId: Option[UUID]) extends Deferred[Int]
 case class DeferFileChecksProgress(consignmentId: Option[UUID]) extends Deferred[FileChecks]
+case class DeferParentFolder(consignmentId: Option[UUID]) extends Deferred[String]
```

Comment:
Now that the parent folder is an optional value, should this deferred resolver also return an optional type? Then you could get rid of the default on line 23.

Reply:
I think that makes much more sense, yeah :smile: